metadata-ingestion/src/datahub/ingestion/source/gc/soft_deleted_entity_cleanup.py

```diff
@@ -274,6 +275,28 @@ def _get_urns(self) -> Iterable[str]:
         )
         yield from self._get_soft_deleted_queries()
 
+    def _times_up(self) -> bool:
+        if (
+            self.config.runtime_limit_seconds
+            and time.time() - self.start_time > self.config.runtime_limit_seconds
+        ):
+            logger.info(
+                f"Runtime limit of {self.config.runtime_limit_seconds} seconds reached"
+            )
+            return True
+        return False
+
+    def _deletion_limit_reached(self) -> bool:
+        if (
+            self.config.limit_entities_delete
+            and self.report.num_hard_deleted > self.config.limit_entities_delete
+        ):
+            logger.info(
+                f"Limit of {self.config.limit_entities_delete} entities reached"
+            )
```

Comment:
We could give more visibility to this by using the structured report.

Reply:
done